spec/system/admin/orders/invoices_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+describe '
+ As an administrator
+ I want to create an invoice for an order
+' do
+  include WebHelper
+  include AuthenticationHelper
+
+  let(:user) { create(:user) }
+  let(:product) { create(:simple_product) }
+  let(:distributor) { create(:distributor_enterprise, owner: user, charges_sales_tax: true) }
+  let(:order_cycle) do
+    create(:simple_order_cycle, name: 'One', distributors: [distributor],
+                                variants: [product.variants.first])
+  end
+
+  let(:order) do
+    create(:order_with_totals_and_distribution,
+           distributor:, user:,
+           order_cycle:, state: 'complete',
+           payment_state: 'balance_due')
+  end
+  let(:customer) { order.customer }
+
+  before do
+    Flipper.enable(:invoices)
+    order.finalize!
+    login_as_admin
+    visit spree.edit_admin_order_path(order)
+  end
+
+  context 'when the order has no invoices' do
+    it 'creates an invoice for the order' do
+      click_link 'Invoices'
+
+      expect {
+        click_link 'New Invoice'
+        sleep 2
```

Comment:
This one was strongly needed? I'd love to avoid as much as possible this kind of `sleep` in the code...

Reply:
what do you think about replacing it with [page.driver.wait_for_network_idle](https://github.com/rubycdp/cuprite/tree/c7dc979350295f29630ad6dba35227c5d8a707fc#network-traffic)? ( thanks @filipefurtad0  for sharing)

Reply:
I think it's better, even if I tend to avoid this as well. But if it's strongly needed, let's do this!

Reply:
I would try to check something UI related instead of waiting, if possible (I am not familiar with this page)